api/proto/teleport/legacy/client/proto/authservice.proto

```diff
@@ -1785,6 +1785,14 @@ message ListResourcesRequest {
   // UsePreviewAsRoles indicates that the response should include all resources
   // the caller would be able to access with their preview_as_roles
   bool UsePreviewAsRoles = 12 [(gogoproto.jsontag) = "use_preview_as_roles,omitempty"];
+  // SSHTargetHost matches ssh server resources as if it were the host argument in a dial request.
+  string SSHTargetHost = 13 [(gogoproto.jsontag) = "ssh_target_host,omitempty"];
+  // SSHTargetPort assists in matching ssh server resources as if they were being targeted for a dial
+  // request. This field has no effect unless SSHTargetHost is also specified.
+  string SSHTargetPort = 14 [(gogoproto.jsontag) = "ssh_target_port,omitempty"];
+  // CaseInsensitiveRouting determines wether or not the hostname parameter should use case-insensitive
+  // routing semantics when matching. This field is set by the auth server, values sent by clients are ignored.
+  bool CaseInsensitiveRouting = 15 [(gogoproto.jsontag) = "case_insensitive_routing,omitempty"];
```

Comment:
Not that I'm aware of.  If there was one it would be _very_ complex, and we'd need to find some way to propagate dns resolution results into the expression, so it'd definitely at least be a breaking change.  More importantly though, we'd have to commit to all routing logic going forward maintaining compatibility with the predicate syntax, otherwise we'd just run right back into the issue of things like ambiguous host errors and resource-based access requests not necessarily working with the same resource set as the one the proxy is working with.

We _could_ make this a separate GRPC method specifically for this purpose (`clt.GetSSHDialMatches(...)` or something along those lines).  I went with adding this to `ListResources` because thats how the old method worked, but I'm open to the idea that this is too niche a use case for the general `ListerResources` method.  WDYT?